packages/security-questions/src/SecurityQuestionStore.ts

```diff
@@ -12,22 +12,26 @@ class SecurityQuestionStore implements ISerializable {
 
   questions: string;
 
-  constructor({ nonce, shareIndex, sqPublicShare, polynomialID, questions }: SecurityQuestionStoreArgs) {
+  keyType: KeyType;
+
+  constructor({ nonce, shareIndex, sqPublicShare, polynomialID, questions, keyType }: SecurityQuestionStoreArgs) {
     this.nonce = new BN(nonce, "hex");
     this.shareIndex = new BN(shareIndex, "hex");
-    this.sqPublicShare = new PublicShare(sqPublicShare.shareIndex, sqPublicShare.shareCommitment);
+    this.keyType = keyType in KeyType ? keyType : KeyType.secp256k1;
+    this.sqPublicShare = new PublicShare(sqPublicShare.shareIndex, sqPublicShare.shareCommitment, this.keyType);
     this.polynomialID = polynomialID;
     this.questions = questions;
   }
 
   static fromJSON(value: StringifiedType): SecurityQuestionStore {
-    const { nonce, shareIndex, sqPublicShare, polynomialID, questions } = value;
+    const { nonce, shareIndex, sqPublicShare, polynomialID, questions, keyType } = value;
     return new SecurityQuestionStore({
       nonce: new BN(nonce, "hex"),
       shareIndex: new BN(shareIndex, "hex"),
       sqPublicShare: PublicShare.fromJSON(sqPublicShare),
       polynomialID,
       questions,
+      keyType,
```

Comment:
I think the `SecurityQuestionStore` dont need the keyType.
it is already passed in during the constructor, so we dont need to store it

Reply:
This is from a JSON string.